src/components/feed-sources/feed-source-manager.jsx

```diff
@@ -0,0 +1,249 @@
+ * @param {boolean} props.isLoading Is the theme state loading?
+ * @param {object} props.loadingError Loading error.
+ * @param props.mode
+ * @returns {object} The theme form.
+ */
+function FeedSourceManager({
+  saveMethod,
+  id = null,
+  isLoading = false,
+  loadingError = null,
+  initialState = null,
+}) {
+  // Hooks
+  const { t } = useTranslation("common", {
+    keyPrefix: "feed-source-manager",
+  });
+  const navigate = useNavigate();
+
+  // State
+  const [headerText] = useState(
+    saveMethod === "PUT" ? t("edit-feed-source") : t("create-new-feed-source")
+  );
+
+  const [loadingMessage, setLoadingMessage] = useState(
+    t("loading-messages.loading-feed-source")
+  );
+
+  const [dynamicFormElement, setDynamicFormElement] = useState();
+  const [submitting, setSubmitting] = useState(false);
+  const [formStateObject, setFormStateObject] = useState();
+
+  const [
+    postV2FeedSources,
+    { error: saveErrorPost, isSuccess: isSaveSuccessPost },
+  ] = usePostV2FeedSourcesMutation();
+
+  const [
+    PutV2FeedSourcesById,
+    { error: saveErrorPut, isSuccess: isSaveSuccessPut },
+  ] = usePutV2FeedSourcesByIdMutation();
+
+  const feedSourceTypeOptions = [
+    {
+      value: "App\\Feed\\EventDatabaseApiFeedType",
+      title: t("dynamic-fields.EventDatabaseApiFeedType.title"),
+      key: "1",
+      template: <EventDatabaseApiFeedTypeTemplate mode={saveMethod} />,
+    },
+    {
+      value: "App\\Feed\\NotifiedFeedType",
+      title: t("dynamic-fields.NotifiedFeedType.title"),
+      key: "2",
+      template: <NotifiedFeedTypeTemplate mode={saveMethod} />,
+    },
+    {
+      value: "App\\Feed\\CalendarApiFeedType",
+      title: t("dynamic-fields.CalendarApiFeedType.title"),
+      key: "3",
+      template: <CalendarFeedTypeTemplate mode={saveMethod} />,
+    },
+    {
+      value: "App\\Feed\\RssFeedType",
+      title: t("dynamic-fields.RssFeedType.title"),
+      key: "5",
+      template: null,
+    },
+  ];
+
+  /** Set loaded data into form state. */
+  useEffect(() => {
+    setFormStateObject({ ...initialState });
+  }, [initialState]);
+
+  useEffect(() => {
+    if (formStateObject && formStateObject.feedType) {
+      let newSecrets = {};
+
+      switch (formStateObject.feedType) {
+        case "App\\Feed\\EventDatabaseApiFeedType":
+          newSecrets =
+            formStateObject.host === "" ? [] : { host: formStateObject.host };
+          break;
+        case "App\\Feed\\NotifiedFeedType":
+          newSecrets =
+            formStateObject.token === ""
+              ? []
+              : { token: formStateObject.token };
+          break;
+        case "App\\Feed\\CalendarApiFeedType":
+          newSecrets =
+            formStateObject.resources === ""
+              ? []
+              : { resources: formStateObject.resources };
+          break;
+        default:
+          break;
+      }
+      formStateObject.secrets = newSecrets;
+    }
+  }, [formStateObject]);
```

Comment:
i think this code should only run if formStateObject.feedType changes. This should be achievable by something along the lines of:

```
useEffect(() => {
  if (formStateObject?.feedType) {
    ... <update secrets>
  }
}, [formStateObject.feedType]);
```

Reply:
I revised this part a bit, but since the formstateobject doesn't support nested keys, this useEffect needs to be called on formstateobject updates, to set e.g. 

formStateObject.token
into
formStateObject.secrets.token